client/my-sites/comments/comment-list/index.jsx

```diff
@@ -132,18 +146,30 @@ export class CommentList extends Component {
 		} );
 	};
 
-	setCommentStatus = ( commentId, postId, status, options = { showNotice: true } ) => {
+	setCommentStatus = ( commentId, postId, status, options = { isUndo: false, persist: false, showNotice: true } ) => {
+		const { isUndo, persist, showNotice } = options;
+
 		// TODO: Replace with Redux getComment()
 		const comment = this.getComment( commentId );
 
 		if ( comment && status === comment.status ) {
 			return;
 		}
 
+		if ( persist ) {
+			this.updatePersistedComments( {
+				...comment,
+				i_like: 'approved' !== status ? false : comment.i_like,
+				status,
+			}, isUndo );
```

Comment:
> if someone updates the comment in another session and Calypso becomes aware of it

This would be an issue anyway though.
Say you're casually browsing the Pending list. Somebody else approves a bunch of comments and Calypso picks those changes up: suddenly you'd be seeing comments disappearing from the list for no apparent reason.

In the persistence case, as you said, there's not a huge risk. At least from a "potential screw up" point of view.
Let me try an example:

- You've just approved a comment:
`realComment.status = "unapproved" → "approved"`
- The comment is then persisting in your "Pending" list, with a fixed value of:
`persistedComment.status = "approved"`
- Somebody else decides to unapprove it:
`realComment.status = "approved" → "unapproved"`
- You, unaware of the fact that the comment has in fact been already unapproved again, decide to undo your approval:
`persistedComment.status = "approved" → "unapproved"`
but the real action would go like this instead:
`realComment.status = "unapproved" → "unapproved"`
No harm done.

Of corse if several people keep modifying the same comment back and forth at some point everything will go out of sync, but that's to be expected anyway. 😄